internal/attestation/renderer/renderer.go

```diff
@@ -172,3 +184,105 @@ func getPassFromTerm(confirm bool) ([]byte, error) {
 	}
 	return pw1, nil
 }
+
+func (ab *AttestationRenderer) getSigner(ctx context.Context) (sigstoresigner.Signer, error) {
+	var (
+		signer sigstoresigner.Signer
+		err    error
+	)
+
+	if ab.signingKeyPath != "" {
+		ab.logger.Debug().Str("path", ab.signingKeyPath).Msg("loading key")
+		signer, err = signature.SignerFromKeyRef(context.Background(), ab.signingKeyPath, getPass)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		// key is not provided, let's create one
+		ab.logger.Info().Msg("key not provided, running in key-less mode")
+		signer, err = ab.keyLessSigner(ctx)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return sigdsee.WrapSigner(signer, "application/vnd.in-toto+json"), nil
+}
+
+type certificateRequest struct {
+	PrivateKey *ecdsa.PrivateKey
+	// CertificateRequestPEM contains the signed public key and the CSR metadata
+	CertificateRequestPEM []byte
+}
+
+type Signer struct {
+	sigstoresigner.SignerVerifier
+	Cert  string
+	Chain []string
+}
+
+func (ab *AttestationRenderer) keyLessSigner(ctx context.Context) (sigstoresigner.Signer, error) {
```

Comment:
I have the feeling that this didn't need to be method but instead a function. Since it seems that in most cases the only thing they use from the parent struct is the logger.

In other words, keyLessSigner can probably return a `ChainloopSigner` concrete type, and this time is the one that implements, `getCret`, `createCertReq`, ... That way eventually we will be able to move this to its own package. WDYT?

Reply:
I've moved everything to its own package, and now the signer is injected into the Renderer. I think it's much cleaner now.